umm-spec-lib/src/cmr/umm_spec/migration/related_url_migration.clj

```diff
@@ -251,3 +336,46 @@
    (update :ContactGroups migrate-contacts-down)
    (update :ContactPersons migrate-contacts-down)
    (update :DataCenters migrate-data-centers-down)))
+
+(defn replace-existing-related-url-keywords
+  "Pass in a related url and UMM version 1.10 or 1.11. Replace the existing UMMM-C version 1.11/1.10
+   with 1.10/1.11 keywords. If a bad version is given just use the original keywords."
+  [related-url toVersion]
+  (let [{:keys [URLContentType Type Subtype]} related-url
+        keywords (remove-nil-keys {:URLContentType URLContentType
+                                   :Type Type
+                                   :Subtype Subtype})
+        result (case toVersion
+                     "1.10" (ru-maps/umm-1-11-umm-url-types->umm-1-10-umm-url-types keywords)
+                     "1.11" (ru-maps/umm-1-10-umm-url-types->umm-1-11-umm-url-types keywords)
+                     keywords)]
+    (remove-nil-keys
+     (assoc related-url :URLContentType (:URLContentType result)
+                        :Type (:Type result)
+                        :Subtype (:Subtype result)))))
+
+(defn migrate-up-to-1_11
+  ":RelatedUrl Type and Subtypes have changed in UMM-C version 1.11 to correspond
+   to GCMD Related URL Keyword updates 8.6."
+  [collection]
+  (def cc collection)
+  (if (nil? (:RelatedUrls collection))
+    collection
+    (assoc collection :RelatedUrls
+      (into []
+       (:RelatedUrls
+        (-> collection
+          (update-in-each [:RelatedUrls] replace-existing-related-url-keywords "1.11")))))))
+
+(defn migrate-down-from-1_11
+  ":RelatedUrl Type and Subtypes have changed in UMM-C version 1.11 to correspond
+   to GCMD Related URL Keyword updates 8.6, so we need to translate the values back to the old
+   8.5 keyword version"
+  [collection]
+  (if (nil? (:RelatedUrls collection))
```

Comment:
Should be able to change to `when` similar to my comment on the migrate up function.

Reply:
I can't.  This function needs to return a collection.  If the when block is false nil is passed back instead of the actual collection.